docs/usage/private-modules.md

```diff
@@ -5,6 +5,120 @@ description: How to support private modules when using Renovate
+- Use `local>....` syntax to refer to private presets
+
+## Dependency Version Lookups
+
+Whenever Renovate detects that a project uses a particular dependency, it attempts to look up that dependency to see if any new versions exist. If such a package is private, then Renovate will need to be configured with the relevant credentials. Renovate does not use any package managers for this step and performs all HTTP(S) lookups itself, including insertion of authentication headers.
+
+Configuring Renovate with credentials requires `hostRules`. Each host rule consists of a `hostType` value and/or a way to match against hosts using `baseUrl`, `hostName` or `domainName`.
+
+`hostType` is not particularly important at this step unless you have different credentials for the same host, however it is sometimes useful in later steps so is good to include if you can. It can be either a "platform" name (e.g. `github`, `azure`, etc) or a "datasource" name (e.g. `npm`, `maven`, `github-tags`, etc).
+
+`baseUrl` can be used if you want to only apply the credentials for a nested path within the host, e.g. `https://registry.company.com/nested/path/`. If the same credentials apply to all paths on a host, then use `hostName` instead, e.g. `registry.company.com`. Finally, to apply credentials to all hosts within the domain, use `domainName`, e.g. `company.com`. You need to pick only one of these and not try to use multiple at the same time, or it will be a config error.
+
+In addition to the above options to match against a host, you need to add the credentials. Typically they are either `token`, or `username` + `password`. Other credential terms are not supported yet.
+
+Here is an example of some host rules:
+
+```json
+{
+  "hostRules": [
+    {
+      "hostName": "registry.npmjs.org",
+      "token": "abc123"
+    },
+    {
+      "baseUrl": "https://registry.company.com/pypi-simple/",
+      "username": "engineering",
+      "password": "abc123"
+    }
+  ]
+}
+```
+
+Renovate applies theses `hostRules` to every HTTP(s) request which is sent, so they are largely independent of any platform or datasource logic. With `hostRules` in place, private package lookups should all work.
+
+## Release Notes
+
+When Renovate creates Pull Requests, its default behavior is to locate and embed release notes/changelogs of packages. These release notes are fetched from the source repository of packages and not from the registries themselves, so if they are private then they will require different credentials.
+
+When it comes to open source, most packages host their source on `github.com` in public repositories. However, GitHub greatly rate limits unauthenticated API requests so there is a need to configure credentials for github.com as otherwise the bot will get rate limited quickly. It can be confusing for people who host their own source code privately to be asked to configure a `github.com` token but without it Release Notes for most open source packages will be blocked.
```

Comment:
```suggestion
When it comes to open source, most packages host their source on `github.com` in public repositories.
However, GitHub greatly rate limits unauthenticated API requests so there is a need to configure credentials for github.com as otherwise the bot will get rate limited quickly.
It can be confusing for people who host their own source code privately to be asked to configure a `github.com` token but without it Release Notes for most open source packages will be blocked.
```